disasm/disasm.cc

```diff
@@ -834,6 +834,34 @@ void disassembler_t::add_instructions(const isa_parser_t* isa)
     DEFINE_XAMO(sc_d)
   }
 
+  if (isa->extension_enabled(EXT_ZAAMO)) {
+    DEFINE_XAMO(amoadd_w)
+    DEFINE_XAMO(amoswap_w)
+    DEFINE_XAMO(amoand_w)
+    DEFINE_XAMO(amoor_w)
+    DEFINE_XAMO(amoxor_w)
+    DEFINE_XAMO(amomin_w)
+    DEFINE_XAMO(amomax_w)
+    DEFINE_XAMO(amominu_w)
+    DEFINE_XAMO(amomaxu_w)
+    DEFINE_XAMO(amoadd_d)
+    DEFINE_XAMO(amoswap_d)
+    DEFINE_XAMO(amoand_d)
+    DEFINE_XAMO(amoor_d)
+    DEFINE_XAMO(amoxor_d)
+    DEFINE_XAMO(amomin_d)
+    DEFINE_XAMO(amomax_d)
+    DEFINE_XAMO(amominu_d)
+    DEFINE_XAMO(amomaxu_d)
+  }
```

Comment:
The code duplication seems unfortunate.  Can we regroup this with the previous `if` statement into `if (isa->extension_enabled(EXT_ZAAMO) || isa->extension_enabled('A'))`?

Reply:
Updated.